src/backend/executor/spi.c

```diff
@@ -1455,6 +1468,15 @@ SPI_cursor_open_internal(const char *name, SPIPlanPtr plan,
 
 	Assert(portal->strategy != PORTAL_MULTI_QUERY);
 
+	gp_enable_gpperfmon = orig_gp_enable_gpperfmon;
```

Comment:
Only PortalStart will be executed with disabled perfmon with these changes. But all other stages (Run,Finish,End) will be proceeded with default value. Is it expected? Does it normally handled by perfmon server side?